js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useStaticSetFilter.tsx

```diff
@@ -94,27 +100,69 @@ export function useStaticSetFilter<TValue>({
             key: getKey?.(value) || index.toString(),
             value,
           }));
+        } else {
+          results = allValues
+            .filter(({match}) =>
+              match.some((value) => value.toLowerCase().includes(query.toLowerCase())),
+            )
+            .map(({value}, index) => ({
+              label: (
+                <SetFilterLabel
+                  value={value}
+                  renderLabel={renderLabel}
+                  filter={filterObjRef.current}
+                  allowMultipleSelections={allowMultipleSelections}
+                />
+              ),
+              key: getKey?.(value) || index.toString(),
+              value,
+            }));
         }
-
-        return allValues
-          .filter(({match}) =>
-            match.some((value) => value.toLowerCase().includes(query.toLowerCase())),
-          )
-          .map(({value}, index) => ({
-            label: (
-              <SetFilterLabel
-                value={value}
-                renderLabel={renderLabel}
-                filter={filterObjRef.current}
-                allowMultipleSelections={allowMultipleSelections}
-              />
-            ),
-            key: getKey?.(value) || index.toString(),
-            value,
-          }));
+        if (allowMultipleSelections) {
+          return [
+            {
+              label: <ButtonLink>Select all</ButtonLink>,
+              key: 'useStaticSetFilter:select-all-key',
+              value: selectAllSymbol,
+            },
+            ...results,
+          ];
+        }
+        return results;
       },
       onSelect: ({value, close}) => {
-        let newState = new Set(filterObjRef.current.state);
+        const state = filterObjRef.current.state;
+        if (typeof value === 'symbol' && value.toString() === selectAllSymbol.toString()) {
+          let selectResults;
+          if (currentQueryRef.current === '') {
+            selectResults = allValues;
+          } else {
+            selectResults = allValues.filter(({match}) =>
+              match.some((value) =>
+                value.toLowerCase().includes(currentQueryRef.current.toLowerCase()),
+              ),
+            );
+          }
+          const hasAnyUnselected = selectResults.find((result) => {
+            if (state instanceof Array) {
+              return state.indexOf(result.value) === -1;
+            } else if (state instanceof Set) {
+              return !state.has(result.value);
+            }
+            return true;
+          });
+          if (hasAnyUnselected) {
+            setState(new Set([...Array.from(state), ...selectResults.map(({value}) => value)]));
```

Comment:
Since we have to convert state to an array anyway, maybe we do it ahead of line 146 to make this logic simpler 👀 

Is there a specific filter this supports arrays for? I wonder if we could upgrade usages of this to be consistently `Set` types to make this less complicated